test/frameworks/karma.test.js

```diff
@@ -0,0 +1,229 @@
+			const error_message2 = 'new error message';
+			const error_stack2 = 'RealException: Error happened again';
+			const suite2 = [
+				'otherFunctionTested',
+				'Better Use Case'
+			];
+
+			const fileInfoStub = sinon.stub(Karma, '_getFileInfo');
+
+			fileInfoStub.onFirstCall().returns({
+				path: filePath1,
+				line: line1
+			});
+
+			fileInfoStub.onSecondCall().returns({
+				path: filePath2,
+				line: line2
+			});
+
+			sinon.stub(Helpers, 'extractJSON').returns({
+				summary: {
+					failed: 2
+				},
+				browsers: {
+					123: {
+						name: browser_name
+					}
+				},
+				result: {
+					123: [
+						{
+							success: true
+						}, {
+							success: false,
+							log: [error_message1, error_stack1],
+							suite: suite1,
+							description: title1
+						}, {
+							success: false,
+							log: [error_message2, error_stack2],
+							suite: suite2,
+							description: title2
+						}
+					]
+				}
+			});
+
+			const result = Karma.parseTestResult('');
+
+			expect(result.title).to.equal(ANNOTATION_TITLE);
+			expect(result.summary).to.equal(`2 ${ANNOTATION_SUMMARY}`);
+			expect(result.annotations.length).to.equal(2);
+
+			expect(result.annotations[0].path).to.equal(filePath1);
+			expect(result.annotations[0].start_line).to.equal(line1);
+			expect(result.annotations[0].end_line).to.equal(line1);
+			expect(result.annotations[0].annotation_level).to.equal('failure');
+			expect(result.annotations[0].message).to.equal(`${suite1.join('.')}: ${title1} (${browser_name}):\n\n${error_message1}\n${error_stack1}`);
+			expect(result.annotations[0].title).to.equal(`${title1} (${browser_name})`);
+
+			expect(result.annotations[1].path).to.equal(filePath2);
+			expect(result.annotations[1].start_line).to.equal(line2);
+			expect(result.annotations[1].end_line).to.equal(line2);
+			expect(result.annotations[1].annotation_level).to.equal('failure');
+			expect(result.annotations[1].message).to.equal(`${suite2.join('.')}: ${title2} (${browser_name}):\n\n${error_message2}\n${error_stack2}`);
+			expect(result.annotations[1].title).to.equal(`${title2} (${browser_name})`);
+		});
+	});
+
+	describe('_getFileInfo', () => {
+		it('Can get line number', () => {
+			const filePath = 'test/somewhere/here.test.js';
+			const line = 42;
+
+			const result = Karma._getFileInfo(`
+			Some error happened!/base/${filePath}:${line}:13
+		`);
+
+			expect(result.path).to.equal(filePath);
+			expect(result.line).to.equal(line);
+		});
+
+		it('Can get line number when no column', () => {
+			const filePath = 'test/somewhere/here.test.js';
+			const line = 42;
+
+			const result = Karma._getFileInfo(`
+			Some error happened!/base/${filePath}:${line}
+		`);
+
+			expect(result.path).to.equal(filePath);
+			expect(result.line).to.equal(line);
+		});
+
+		it('Defaults to line 1 of package.json if no file', () => {
+			const result = Karma._getFileInfo(`
+			Some error happened!/base/test/somewhere/here.js:42:13

```

Comment:
Is this test for if the error happens oustide of a `.test` file?

Reply:
The stack given by karma can contain references to multiple files. I am going with the assumption that I only care about the test.js file since it is my only way of guessing the test file the error has occurred. If I just took the first `js` file that shows up in the stack I would end up with a node_modules file